RecoLocalCalo/HGCalRecAlgos/plugins/HGCalConfigurationESProducer.cc

```diff
@@ -0,0 +1,231 @@
+#include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
+#include "FWCore/Framework/interface/ESProducts.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+#include "FWCore/Utilities/interface/do_nothing_deleter.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CondFormats/HGCalObjects/interface/HGCalConfiguration.h"
+#include "CondFormats/HGCalObjects/interface/HGCalMappingModuleIndexer.h"
+#include "CondFormats/DataRecord/interface/HGCalElectronicsMappingRcd.h"
+#include "CondFormats/DataRecord/interface/HGCalModuleConfigurationRcd.h"  // depends on HGCalElectronicsMappingRcd
+#include <string>                                                          // for std::to_string
+#include <fstream>  // needed to read json file with std::ifstream
+#include <nlohmann/json.hpp>
+using json = nlohmann::json;
+
+/**
+   @short plugin parses HGCAL electronics configuration from JSON file
+ */
+class HGCalConfigurationESProducer : public edm::ESProducer, public edm::EventSetupRecordIntervalFinder {
+public:
+  explicit HGCalConfigurationESProducer(const edm::ParameterSet& iConfig)
+      :  //edm::ESProducer(iConfig),
+        fedjson_(iConfig.getParameter<std::string>("fedjson")),
+        modjson_(iConfig.getParameter<std::string>("modjson")) {
+    if (iConfig.exists("bePassthroughMode"))
+      bePassthroughMode_ = iConfig.getParameter<int32_t>("bePassthroughMode");
+    if (iConfig.exists("econPassthroughMode"))
+      econPassthroughMode_ = iConfig.getParameter<int32_t>("econPassthroughMode");
+    if (iConfig.exists("cbHeaderMarker"))
+      cbHeaderMarker_ = iConfig.getParameter<int32_t>("cbHeaderMarker");
+    if (iConfig.exists("slinkHeaderMarker"))
+      slinkHeaderMarker_ = iConfig.getParameter<int32_t>("slinkHeaderMarker");
+    if (iConfig.exists("econdHeaderMarker"))
+      econdHeaderMarker_ = iConfig.getParameter<int32_t>("econdHeaderMarker");
+    if (iConfig.exists("charMode"))
+      charMode_ = iConfig.getParameter<int32_t>("charMode");
+    if (iConfig.exists("gain"))
+      gain_ = iConfig.getParameter<int32_t>("gain");
+    auto cc = setWhatProduced(this);
+    //findingRecord<HGCalModuleConfigurationRcd>();
+    indexToken_ = cc.consumes(iConfig.getParameter<edm::ESInputTag>("indexSource"));
+  }
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::ESInputTag>("indexSource", edm::ESInputTag(""))
+        ->setComment("Label for module indexer to set SoA size");
+    //desc.add<edm::FileInPath>("filename")->setComment("JSON file with FED configuration parameters");
+    desc.add<std::string>("fedjson", "")->setComment("JSON file with FED configuration parameters");
+    desc.add<std::string>("modjson", "")->setComment("JSON file with ECOND configuration parameters");
+    desc.addOptional<int32_t>("bePassthroughMode", -1)
```

Comment:
These parameters should normally be taken from JSON files. I implemented these parameters such that users can exceptionally override the JSON for for quick debugging during system tests (scans in the lab or test beams), instead of editing the JSONs. I made them optional so the user is not not required to explicitly set them in the python cfg file. If it is okay with you I prefer to keep it as it is for now?

> In addition, if these parameters are not set, the corresponding member variables are left uninitialized. Their use in `getint()` and `gethex()` to override the other argument of the function if they are non-negative begs for random (well, strictly speaking undefined) behavior.

They are now initialized as -1.